### DISCUSSION TITLE
How to Retrieve the Basis Order of a Setup?

### DISCUSSION THREAD

**Discussion by chamanth-vct** [ansys/pyaedt#5411] [Q&A] [answered]
Hi, 

I need to find the order of basis function of a setup using Pyaedt. I have managed to access the setup with the following code:

```py
desktop = ansys.aedt.core.Desktop() 
prj = desktop.load_project(file_path)
setup = prj.setup[0]
```

Now, I am trying to find out how to retrieve the order of the basis function for this setup. Is there a specific attribute or method in PyAEDT that can provide this information?

<img src="https://github.com/user-attachments/assets/408a318f-7e7a-4833-bb6d-6aeb7925709b" width="400">

Any guidance or references would be greatly appreciated. Thank you!

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @chamanth-vct ,

You can access all setup props:
![image](https://github.com/user-attachments/assets/4d9a6fe5-af4b-4e91-9b43-21dff17733a9)


And you just need to set the property:
![image](https://github.com/user-attachments/assets/6f19ed1c-0334-4426-9893-9670ede28881)


Mapping all setup properties is challenging, because it is shared between all AEDT apps (HFSS, Icepak, Maxwell...), so this is the best way to update the setup properties.

**Reply by luiGarcig** [reply]
Hello,

so, if i want to set this property with "Mixed Order", how can i do that? 

I am using a function like this to configure my setup:

```
def SingleFrequency(
    hfss,
    setup,
    freq,
    maxPasses,
    maxDeltaS= 0.01,
    name="MySetup"
):
    setup.props["Frequency"] = freq
    setup.props["MaximumPasses"] = maxPasses
    setup.props["MaxDeltaS"] =maxDeltaS
    setup.props["MinimumPasses"] = 2
    setup.props["MinimumConvergedPasses"] = 2
    setup.props["UseIterativeSolver"] = True
    setup.props["SaveRadFieldsOnly"] = True
    setup.props["RadiationSetup"] = "3D"
    setup.props["ComputeFarFields"] = True
    hfss.setups[0].props["BasisOrder"] = ??
```
